src/s3/commands/presignedURL.ts

```diff
@@ -38,36 +45,124 @@ export async function presignedURLCommand(
 
         await copyToClipboard(url, 'URL', window, env)
+        vscode.window.showInformationMessage(`A presigned URL has been copied to your clipboard`)
     })
 }
 
-export async function promptTime(fileName: string, window = Window.vscode()): Promise<number> {
-    const timeStr = await window.showInputBox({
-        value: '15',
-        prompt: localize(
-            'AWS.s3.promptTime.prompt',
-            'Specify the time (minutes) until URL will expire for path: {0}',
-            fileName
-        ),
-        placeHolder: localize('AWS.s3.promptTime.placeHolder', 'Defaults to 15 minutes'),
-        validateInput: validateTime,
-    })
+function validateTime(time: string): string | undefined {
+    const number = Number(time)
+
+    if (isNaN(Number(time)) || !Number.isSafeInteger(number) || Number(time) <= 0) {
+        return invalidNumberWarning
+    }
+
+    return undefined
+}
+
+export interface PresignedUrlWizardState {
+    region: string
+    signedUrlParams: SignedUrlRequest
+}
+
+export class PresignedUrlWizard extends Wizard<PresignedUrlWizardState> {
+    constructor(node?: S3FileNode) {
+        super({
+            initState: {
+                region: node?.bucket.region,
+                signedUrlParams: node ? ({ bucketName: node.bucket.name } as SignedUrlRequest) : undefined,
+            },
+        })
+        this.form.region.bindPrompter(() => createRegionPrompter().transform(region => region.id))
+        this.form.signedUrlParams.operation.bindPrompter(() => createOperationPrompter())
+        this.form.signedUrlParams.bucketName.bindPrompter(({ region }) =>
+            createBucketPrompter(assertDefined(region, 'region'))
+        )
+
+        if (node) {
+            this.form.signedUrlParams.key.setDefault(node.file.key)
+            this.form.signedUrlParams.key.bindPrompter(
+                ({ region, signedUrlParams }) =>
+                    createS3FilePrompter(
+                        assertDefined(region, 'region'),
+                        assertDefined(signedUrlParams?.bucketName, 'bucketName'),
+                        assertDefined(signedUrlParams?.operation, 'operation')
+                    ),
+                { showWhen: state => state.signedUrlParams?.operation === 'putObject' }
+            )
+        } else {
+            this.form.signedUrlParams.key.bindPrompter(({ region, signedUrlParams }) =>
+                createS3FilePrompter(
+                    assertDefined(region, 'region'),
+                    assertDefined(signedUrlParams?.bucketName, 'bucketName'),
+                    assertDefined(signedUrlParams?.operation, 'operation')
+                )
+            )
+        }
 
-    const time = Number(timeStr)
+        this.form.signedUrlParams.time.bindPrompter(() => createExpiryPrompter().transform(s => Number(s) * 60))
 
-    if (isNaN(time) || time < 0) {
-        throw new ToolkitError('The provided expiration time is not a positive number', { code: 'InvalidInput' })
```

Comment:
old code raised `ToolkitError... code:InvalidInput`. What does the new code raise? Can this affect telemetry and possibly control flow ?

Reply:
Nope, there is a validator on this prompt called `validateTime`. The user cannot progress past this prompt unless they put in a valid positive number.